src/basic/parse-util.c

```diff
@@ -566,6 +566,36 @@ int safe_atod(const char *s, double *ret_d) {
         return 0;
 }
 
+int safe_hexstring_aton(const char *s, uint8_t *b, unsigned int buf_len, unsigned int *len) {
+        unsigned int c = 0;
+        char *p;
+
+        assert(s);
+        assert(b);
+        assert(len);
+
+        if (strlen(s) % 2)
+                return -EINVAL;
+
+        for(c = 0; c < buf_len && strlen(s) > 1; s += 2) {
+                unsigned int i;
+                char t[3] = {};
+
+                strncpy(t, s, 2);
```

Comment:
Paraphrasing strncpy manual page: if there is no null byte among the first 2 bytes of s, the string placed in t will not be null-terminated. Please add t[2] = '\0'.

Reply:
would that be necessary it's initialized to `'\0'` any copying 2 bytes only or I am wrong .